src/schema.c

```diff
@@ -93,14 +93,36 @@ ldap_schema_new(TALLOC_CTX *ctx)
 LDAPObjectClass**
 ldap_schema_object_classes(const ldap_schema_t *schema)
 {
-    if (!schema)
+    if (!schema || !schema->object_classes_by_oid)
```

Comment:
Consider separating checks of `schema` and `object_classes_by_oid`. If we have separate checks for each case, we can accurately say what exactly the error is.